vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -2762,6 +2762,17 @@ protected void updateSelectionModeOnClient() {
                 selectionMode.name());
     }
 
+    /**
+     * Returns the selection mode for this grid.
+     *
+     * @return the selection mode, not null
+     */
+    public SelectionMode getSelectionMode() {
```

Comment:
Please add a test to `GridSelectionModeTest`, for example:
- Test that initial selection mode reports single
- Changing selection mode reports the respective mode